src/test/java/friday/TaskListTest.java

```diff
@@ -0,0 +1,55 @@
+package friday;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertNull;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import friday.task.Task;
+import friday.task.Todo;
+import friday.util.TaskList;
+
+/**
+ * Tests the functionality of the TaskList class.
+ * Verifies that tasks can be added, deleted, and that the TaskList behaves correctly with different indices.
+ */
+public class TaskListTest {
+
+    private TaskList taskList;
+
+    /**
+     * Sets up the test environment by creating a TaskList object
+     * and adding some sample Todo tasks before each test is run.
+     */
+    @BeforeEach
+    public void setUp() {
+        taskList = new TaskList();
+        taskList.addTask(new Todo("Task 1"));
+        taskList.addTask(new Todo("Task 2"));
+    }
+
+    /**
+     * Tests the deletion of a task with a valid index.
+     * Verifies that the task is correctly removed from the list and that the size of the list is updated.
+     */
+    @Test
+    public void testDeleteTaskWithValidIndex() {
```

Comment:
This works, but you could also try something like deleteTask_validIndex_success?